lib/spack/external/_pytest/config.py

```diff
@@ -1032,29 +1032,10 @@ def _consider_importhook(self, args):
         self._warn_about_missing_assertion(mode)
 
     def _mark_plugins_for_rewrite(self, hook):
```

Comment:
I think this function may as well be removed entirely: it is used only in `_consider_importhook` (in the same module). In fact, *that* function ends up doing nothing given that this function does nothing.

I think what you're saying is that we don't use assertion rewriting, so this logic isn't necessary. I'd prefer if you throw an exception in `_consider_importhook` when `mode == 'rewrite'` instead; also remove `_mark_plugins_for_rewrite` entirely, or throw an exception if it is called.

Reply:
This doesn't work.  `rewrite` is the default -- we just don't want rewriting to affect plugins, as we do not use plugins and we don't want to try to iterate through them using `setuptools`.  Making this a no-op is pretty simple and doesn't add new & complex logic to what should be a very simple patch to remove some functionality.

Reply:
OK, can you remove the `_mark_plugins_for_rewrite` function entirely? Or throw an exception if it is called? That will clarify that we don't expect any other part of the code to use the function.

Reply:
Given that nothing calls it I do not see why this is needed -- it makes it harder to update the code if we upgrade our `pytest` version.

Also the function will already raise an exception if it is called -- as it imports `pkg_resources`.  This is no worse than what already happens.